### ISSUE TITLE
autoformat on save

### ISSUE THREAD

**Issue by alixander** [andorsk/d2-mode#30]
Latest D2 version exposes a command for autoformatting D2 scripts. Intended to be used like Go's, i.e. format on save

https://github.com/terrastruct/d2/pull/292

**Comment by andorsk** [comment]
yes! good callout. thanks @alixander

**Comment by avelino** [comment]
Example:
- [`gofmt`](https://github.com/dominikh/go-mode.el/blob/166dfb1e090233c4609a50c2ec9f57f113c1da72/go-mode.el#L1903-L1962)
- [`gofmt-before-save`](https://github.com/dominikh/go-mode.el/blob/166dfb1e090233c4609a50c2ec9f57f113c1da72/go-mode.el#L1997-L2005)

**Comment by andorsk** [comment]
@avelino yep. I'm aware of hooks you can configure on save with a hook. I think again, the tricky part here is region support. If that's not important for most people, I'm happy to implement a quick format command on the d2-mode. Personally, I tend to use d2 mostly in regions, which is why I've held back on implementing the simple solution. 

Proposal here: 

1. Implement format support with caveat that region's won't be supported for now. 
2. Region formatting will be a roadmap item. I'm guessing it will be must more involved, as we'll have to write something to scan the file and format appropriately. Not sure what the right solution is there. 

Are @avelino and @alixander ok with excluding region support?

**Comment by alixander** [comment]
Is regions like a markdown block? I do mostly edit just the d2 files standalone.

**Comment by andorsk** [comment]
@alixander exactly. Regions are blocks.

As an example:

This is a graph

\```d2
A --> B
\```
Commentary here.

Another graph: 

\```d2
A --> B
\```

Commentary here. 

I mostly do graphs in regions,  so for me ( personally ) unless there's region support it becomes a lot less useful.

**Comment by andorsk** [comment]
Started work on a PR here: https://github.com/andorsk/d2-mode/pull/37
